ocpp-v1_6/src/main/java/eu/chargetime/ocpp/JSONClient.java

```diff
@@ -67,8 +67,9 @@ public JSONClient(ClientCoreProfile coreProfile, String identity) {
         featureRepository.addFeatureProfile(coreProfile);
     }
 
-    public void enableWSS(SSLContext sslContext) throws IOException {
-        transmitter.enableWSS(sslContext);
+    public JSONClient enableWSS(WssSocketBuilder wssSocketBuilder) throws IOException {
```

Comment:
It acts like a builder. Not that I mind that much, but why?

Reply:
The reason behind is that we have creation of client and initialization of connection separated in the logic flow. A t the same time we need to ensure some parameters for a socket creation are present at the client creation time (SSLSocketFactory) and some of them we know only at the moment when we initialize connection (URI). Therefore I used builder to handle that distributed parameters gathering and encapsulate a creation logic at the same time.